src/goto-instrument/cover.cpp

```diff
@@ -1248,23 +1238,22 @@ void instrument_cover_goals(
           i_it->make_assertion(d);
           i_it->source_location=source_location;
           i_it->source_location.set_comment(comment_t);
-          i_it->source_location.set(ID_coverage_criterion, coverage_criterion);
-          i_it->source_location.set_property_class(property_class);
+          i_it->source_location.set_property_class("coverage");
 
           const std::string comment_f="decision `"+d_string+"' false";
           goto_program.insert_before_swap(i_it);
           i_it->make_assertion(not_exprt(d));
           i_it->source_location=source_location;
           i_it->source_location.set_comment(comment_f);
-          i_it->source_location.set(ID_coverage_criterion, coverage_criterion);
-          i_it->source_location.set_property_class(property_class);
+          i_it->source_location.set_property_class("coverage");
         }
         
         for(std::size_t i=0; i<decisions.size()*2; i++)
           i_it++;
       }
       break;
-      
+
+    case coverage_criteriont::BOUNDARY:
     case coverage_criteriont::MCDC:
```

Comment:
Unfortunately, the 'boundary-value-analysis' here is not really independent from the MC/DC criterion. It applies to the test cases generated by MC/DC.

For example, given a the decision 'x<3 && y<=5', three tests will be generated following MC/DC:
1) x<3 && y<=5
2) x>=3 && y<=5
3) x<3 && y>5
The boundary value analysis then applies to each test generated, the 2nd one (similar for the 1st one) will be split into 
2.1) x>3 && y<5
2.2) x==3 && y<5
2.3) x>3 && y==5
2.4) x==3 && y==5 

Is it possible to 
1) create a criterion named (let's say) 'MCDC_BOUNDARY' 
or 2) add an option 'boundary-values' to --cover and its combination with 'mcdc' triggers the boundary value analysis after normal MC/DC?